doc/man7/passphrase-encoding.pod

```diff
@@ -0,0 +1,182 @@
+be misinterpreted as the perfectly valid UTF-8 encoded code point U+00EF (LATIN
+SMALL LETTER I WITH DIARESIS) I<if the passphrase doesn't contain anything that
+would be invalid UTF-8>.
+A pass phrase that contains this kind of byte sequence will give a different
+outcome in OpenSSL 1.1.0 and newer than in OpenSSL older than 1.1.0.
+
+ 0x00 0xC3 0x00 0xAF                    # OpenSSL older than 1.1.0
+ 0x00 0xEF                              # OpenSSL 1.1.0 and newer
+
+On the same accord, anything encoded in UTF-8 that was given to OpenSSL older
+than 1.1.0 was misinterpreted as ISO-8859-1 sequences.
+
+=head2 OSSL_STORE
+
+L<ossl_store(7)> acts as a general interface to access all kinds of objects,
+potentially protected with a pass phrase, a PIN or something else.
+This API currently doesn't stipulate any specific encoding of pass phrases, but
+uses the underlying routines with their behaviours.
+This means that when using the built-in C<file:> scheme loader, the pass phrase
+to unlock a PKCS#12 file will be treated as described for PKCS#12 above, and
+the pass phrase for a PEM files will be treated as the general case described
+above, since that loader uses the same underlying routines.
+I<Note that other loaders will have their own behaviours>.
+
+=head1 RECOMMENDATIONS
+
+This section assumes that you know what pass phrase you should type in is, but
+that it may have been used in a different character encoding than your current
+environment gives you by default.
+For example, the pass phrase may have been used at a time when your default
+encoding was ISO-8859-1 (i.e. "naïve" resulting in the byte sequence 0x6E 0x61
+0xEF 0x76 0x65), and you're now in an environment where your default encoding
+is UTF-8 (i.e. "naïve" resulting in the byte sequence 0x6E 0x61 0xC3 0xAF 0x76
+0x65).
+Whenever it's mentioned that you should use a certain character encoding, it
+should be understood that you either change your environment to use the
+mentioned encoding by default when you type in your pass phrase, or use some
+suitable tool to convert your pass phrase from your default encoding to the
+target encoding.
+
+Also note that the sub-sections below discuss human readable pass phrases.
+This is particularly relevant for PKCS#12 objects, where human readable pass
+phrases are assumed.
+For other objects, it's as legitimate to use any byte sequence (such as a
+sequence of bytes from `/dev/urandom` that's been saved away), which makes any
+character encoding discussion irrelevant; in such cases, simply use the same
+byte sequence as it is.
+
+=head2 Creating new objects
+
+For creating new pass phrase protected objects, make sure the pass phrase is
+encoded using UTF-8.
+This is default on most modern Unixes, but may involve an effort on other
+platforms.
+Specifically for Windows, setting the environment variable
+C<OPENSSL_WIN32_UTF8> will have anything prompted on standard input converted
+to UTF-8.
```

Comment:
"prompted"? Isn't "prompt" rather something one actually reads? While password is something that is entered and not shown, so it's kind of never read. There also is ambiguity in "on standard input". C programmer would immediately think of stdin, and this kind of suggested that you'd use fgets. But that's not what actually happens. You bypass C standard methods and turn straight to Windows for raw UNICODE. So that "will have anything entered on [Windows] console prompt converted to UTF-8" would describe it more adequately.

Just in case for reference. The variable affects treatment of command line arguments, i.e. command line arguments will be encoded as UTF-8, as well as output, i.e. UTF-8 strings emitted to [Windows] console will be rendered correctly.

Reply:
Right, a 'prompt' is printed by the shell as 'invitation' for the user to enter input.

Reply:
Ok, I see the confusion here...  in this case, I was thinking of password prompting, but right, one can pass a password on the command line as well, right.

Re "prompt", I thought it could be used as a verb...  "to prompt for input" (in this case a pass phrase).  Am I wrong?

Reply:
No, you're not wrong. But you are saying "anything prompted on standard input [is] converted to UTF-8". But "anything prompted" normally refers to what the shell prints (i.e., the dollar sign), not what the user enters.

Reply:
Errrr, if not given on the command line somehow, `openssl` will prompt for a pass phrase when it needs one.  So no, I'd say that the command line prompt isn't *the only* prompt.  We use this terminology in `doc/man1/openssl.pod`

Reply:
However, since `OPENSSL_WIN32_UTF8` does enable conversion of command line parameters as well, it's quite right, like @dot-asm says, that this explanation should be limited to the password prompt

Reply:
> "prompt" as  a verb

And "everything you are prompted to enter" would have worked (at least in my mind), because this formulation would refer to your input, and not act of prompting itself.